src/api/app/models/workflow.rb

```diff
@@ -1,12 +1,26 @@
 class Workflow
   SUPPORTED_STEPS = { 'branch_package' => Workflow::Step::BranchPackageStep }.freeze
+  SUPPORTED_FILTERS = [:architectures, :repositories].freeze
+  # The order of the filter types determines their precedence
+  SUPPORTED_FILTER_TYPES = [:only, :ignore].freeze
 
   def initialize(workflow:, scm_extractor_payload:, token:)
-    @workflow = workflow
+    @workflow = workflow.with_indifferent_access
     @scm_extractor_payload = scm_extractor_payload
     @token = token
+
     raise Token::Errors::InvalidWorkflowStepDefinition, "Invalid workflow step definition: #{errors.to_sentence}" unless
       valid?
+
+    # Filters aren't mandatory in a workflow
+    return unless @workflow.key?(:filters)
+
+    raise Token::Errors::UnsupportedWorkflowFilters, "Unsupported filters: #{@unsupported_filters.keys.to_sentence}" if unsupported_filters?
```

Comment:
You can use validations outside or active record too. See https://api.rubyonrails.org/classes/ActiveModel/Validations.html

Reply:
Yes, this is what I had in mind when writing this in the PR description: `We could gather all errors together before raising if any error is present.`